### DISCUSSION TITLE
Barcode registration discrepancy between provided barcode coordinates and generated coordinates from raw data

### DISCUSSION THREAD

**Discussion by StevenSong** [rajewsky-lab/openst#25] [Q&A] [answered]
Thank you for providing the fastq files from barcode registration! I've been trying to replicate the barcode registration processing step where you turn the reads from the fastq into the barcode coordinate matrices. Specifically, I've been comparing outputs generated from the raw data at [fc_2_L3.fastq.gz](http://bimsbstatic.mdc-berlin.de/rajewsky/openst-public-data/fc_2_L3.fastq.gz) to the provided output tile barcode coordinate matrices in [adult_hippocampus_tiles.tar.xz](http://bimsbstatic.mdc-berlin.de/rajewsky/openst-public-data/adult_hippocampus_tiles.tar.xz).

After running `barcode_preprocessing`, I think I've deduced that the barcodes in the provided barcode coordinates are cropped to indices `[5, 30)` or `5:30` in slice notation and are the reverse complement. This makes sense given the cropping scheme defined in the spacemake config for the later sequencing output: https://github.com/rajewsky-lab/spacemake/blob/cdcfc4375b75caeb19fb057a2739ef85359b3d2c/spacemake/data/config/config.yaml#L117-L120

It seems that `barcode_preprocessing` simply extracts reads and their positions with some sequence processing. My question is how are the barcode coordinates post processed? What I've observed from comparing the provided barcode coordinates to the ones I can generate from the fastq is that there's removal of duplicates and some other barcodes to convert the generated output (mine) to the expected output (yours).

EDIT: Apologies for the plethora of edits on this post, I've removed the attempt at illustrating the observations as I'm realizing that the overlap between different subsets of in-tile and cross-tile barcodes is potentially very complex. The core of this question is simply:
* What post-processing on the barcode coordinates are applied?
* Why not keep only truly-unique barcodes (and remove all instances of any duplicated barcode) for a given capture area?

**Comment by StevenSong** [comment]
I've uploaded my generated files from `barcode_preprocessing` for here: https://drive.google.com/file/d/1G4zJey1AtjtFXF4wiCMZdht1HkwkTuny/view?usp=share_link

The above observations are based on the differences between these files and their corresponding files in [adult_hippocampus_tiles.tar.xz](http://bimsbstatic.mdc-berlin.de/rajewsky/openst-public-data/adult_hippocampus_tiles.tar.xz)

**Reply by danilexn** [reply, marked as answer]
Hi Steven! Thanks for the questions :)


>  What post-processing on the barcode coordinates are applied?

There's no post-processing of the barcode coordinates

>  Why not keep only truly-unique barcodes (and remove all instances of any duplicated barcode) for a given capture area?

The number of collisions is very low, but we're also working on implementing HD==1 for error correction. We will extend `barcode_processing.py` to include different duplicate-removal strategies.
